straxen/plugins/led_cal/led_calibration.py

```diff
@@ -122,48 +178,130 @@ def get_records(raw_records, baseline_window):
+        last = hit["record_i"]
+
+    return windows
+
+
 _on_off_dtype = np.dtype([("channel", "int16"), ("amplitude", "float32")])
 
 
-def get_amplitude(records, led_window, noise_window):
+@numba.jit(nopython=True)
+def get_amplitude(records, led_windows, noise_window):
     """Needed for the SPE computation.
 
-    Take the maximum in two different regions, where there is the signal and where there is not.
+    Get the maximum of the signal in two different regions, one where there is no signal, and one
+    where there is.
+
+    :param records: Array of records
+    :param ndarray led_windows : 2d array of shape (len(records), 2) with the window to use as the
+        signal on area for each record. Inclusive left boundary and exclusive right boundary.
+    :param tuple noise_window: Tuple with the window, used for the signal off area for all records.
+    :return ndarray ons: 1d array of length len(records). The maximum amplitude in the led window
+        area for each record.
+    :return ndarray offs: 1d array of length len(records). The maximum amplitude in the noise area
+        for each record.
 
     """
-    on = np.zeros((len(records)), dtype=_on_off_dtype)
-    off = np.zeros((len(records)), dtype=_on_off_dtype)
-    on["amplitude"] = np.max(records["data"][:, led_window[0] : led_window[1]], axis=1)
-    on["channel"] = records["channel"]
-    off["amplitude"] = np.max(records["data"][:, noise_window[0] : noise_window[1]], axis=1)
-    off["channel"] = records["channel"]
-    return on, off
+    ons = np.zeros(len(records), dtype=_on_off_dtype)
+    offs = np.zeros(len(records), dtype=_on_off_dtype)
+
+    for i, record in enumerate(records):
+        ons[i]["channel"] = record["channel"]
+        offs[i]["channel"] = record["channel"]
+
+        ons[i]["amplitude"] = np.max(record["data"][led_windows[i, 0] : led_windows[i, 1]])
+        offs[i]["amplitude"] = np.max(record["data"][noise_window[0] : noise_window[1]])
+
+    return ons, offs
 
 
 _area_dtype = np.dtype([("channel", "int16"), ("area", "float32")])
 
 
-def get_area(records, led_window):
+@numba.jit(nopython=True)
+def get_area(records, led_windows):
     """Needed for the gain computation.
 
-    Sum the data in the defined window to get the area. This is done in 6 integration window and it
-    returns the average area.
+    Integrate the record in the defined window area. To reduce the effects of the noise, this is
+    done with 6 different window lengths, which are then averaged.
+
+    :param records: Array of records
+    :param ndarray led_windows : 2d array of shape (len(records), 2) with the window to use as the
+        integration boundaries.
+    :return ndarray area: 1d array of length len(records) with the averaged integrated areas for
+        each record.
 
     """
-    left = led_window[0]
-    end_pos = [led_window[1] + 2 * i for i in range(6)]
+    area = np.zeros(len(records), dtype=_area_dtype)
+    end_pos = [2 * i for i in range(6)]
```

Comment:
Same here - probably it's better to make a URLConfig for it (something like `max_integration_window`)

Reply:
fixed